tern/classes/image_layer.py

```diff
@@ -106,6 +108,17 @@ def os_guess(self):
     def os_guess(self, os_guess):
         self.__os_guess = os_guess
 
+    @property
+    def files_analyzed(self):
+        return self.__files_analyzed
+        
+    @files_analyzed.setter
+    def files_analyzed(self, x):
+        if bool(x):
```

Comment:
`bool(x)` casts the input `x` to a boolean value, so this will always be assigned.
Instead do: `if isinstance(x, bool):`

Reply:
Thanks a lot for pointing this out @nishakm